src/test/java/htsjdk/samtools/util/IoUtilTest.java

```diff
@@ -170,6 +175,22 @@ public void testFileTypeUnix(final String path, boolean expectedIsRegularFile) {
         Assert.assertEquals(IOUtil.isRegularPath(file), expectedIsRegularFile);
     }
 
+    @Test
+    public void testAddExtension() throws IOException {
+        Path p = IOUtil.getPath("/folder/file");
+        List<FileSystemProvider> fileSystemProviders = FileSystemProvider.installedProviders();
+        Assert.assertEquals(IOUtil.addExtension(p, ".ext"), IOUtil.getPath("/folder/file.ext"));
+        p = IOUtil.getPath("folder/file");
+        Assert.assertEquals(IOUtil.addExtension(p, ".ext"), IOUtil.getPath("folder/file.ext"));
+        FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix());

```

Comment:
The jmfs should be closed. I guess that try-with-resources is the better option for that...